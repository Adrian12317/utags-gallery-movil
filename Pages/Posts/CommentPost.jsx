import React, { useEffect, useState } from 'react'
import { Text,StyleSheet,Alert , View,ScrollView } from 'react-native';
import { Card , Button } from 'react-native-elements'
import {
  StyledContainerStart,
  StyledInputCreate
} from './styledPostsComponent';
import Icon from 'react-native-vector-icons/FontAwesome';
import axios from 'axios';
import ImagePreview from 'react-native-image-preview';
import i18n from "../../localization/i18n"

export default function CommentPost({setIsComment,post_id,getItems}) {
    const [isLoading,setIsLoading] = useState(false)
    const [post,setPost] = useState({})
    const [myComment,setMyComment] = useState()
    const [comments,setComments] = useState([])
    const [isVisible,setIsVisible] = useState(false);
    const [previewUri,setPreviewUri] = useState();
    const imageDefault = "https://www.smartdatajob.com/images/joomlart/demo/default.jpg";
    
    const getPost = async () =>{
        // console.log(post_id);
      await  axios.get('https://api-utagsgallery-codes.herokuapp.com/posts/'+post_id)
      .then(function ({data:{data}}) {
        // handle success
       
        setPost({...data[0]})
      })
      .catch(function (error) {
        // handle error
        setPost({})
        console.log(error);
      })
    }
    const getComments = async () =>{
        console.log(post_id);
      await  axios.get('https://api-utagsgallery-codes.herokuapp.com/comments/'+post_id)
      .then(function ({data}) {
        // handle success
        
       setComments([...data])
        
      })
      .catch(function (error) {
        // handle error
        setComments([])
        console.log(error);
      })
    }

   useEffect(()=>{
    getPost()
    getComments()
   },[])


   const handleComment = async () =>{
    if (!myComment) {return alert(i18n.t("POST_COMMENT").POST_COMMENT_ALERT_EMPTY)}

    let comment = {
      user_id: 72,
      post_id: post.post_id,
      comment: myComment
    }

    await axios.post('https://api-utagsgallery-codes.herokuapp.com/comments',comment)
        .then(function ({data}) {
          // handle success
          setMyComment((current) => current="");
          getComments()

        })
        .catch(function (error) {
          // handle error
          console.log(error);
            return alert("We have an error");
          
        })
   }

   const handlePreview = (img_uri) =>{
    setPreviewUri(img_uri);
    setIsVisible(true);
  }



  const handleDelete = (post_id) =>
  Alert.alert(i18n.t("POST_COMMENT").POST_COMMENT_DELETE_TITLE, i18n.t("POST_COMMENT").POST_COMMENT_DELETE, [
    {
      text: 'Cancel',
      onPress: () => console.log('Cancel Pressed'),
      style: 'cancel',
    },
    { text: 'OK', onPress: () => {
      console.log('OK Pressed')

      axios.put('https://api-utagsgallery-codes.herokuapp.com/posts/delete/'+post_id)
        .then(function ({data}) {
          // handle success
          setIsComment(false);
          getItems()

        })
        .catch(function (error) {
          // handle error
          console.log(error);
            return alert("We have an error");
          
        })
      
    } },
  ]);

  

    return(
        <StyledContainerStart>
          <ImagePreview visible={isVisible} source={{uri: previewUri}} close={()=>{setIsVisible(false)}} />
             <Card >
             <Icon style={{fontSize:16,color:"black"}} name='arrow-left' onPress={()=>setIsComment(false)} />
            <Card.Title>{post.title}</Card.Title>
            <Card.Divider/>
            <Card.Image onPress={()=>handlePreview(post.image_url)} style={styles.image} source={post.image_url? {uri:`${post.image_url}`}:{uri:imageDefault}}>             
            </Card.Image>
            <Text style={{marginBottom: 10}}>
               {post.content}
            </Text>
            <Card.Divider />
            <Text style={{marginBottom: 10}}>
              {i18n.t("POST_COMMENT").POST_COMMENT_YOUR_COMMENT}
            </Text>
            <StyledInputCreate
            multiline={true}
            numberOfLines={2}
            value={myComment}
            onChangeText={setMyComment}
            ></StyledInputCreate>
            <Button
                icon={<Icon style={styles.buttonLike} name='comments' />}
                buttonStyle={{width:120}}
                title={i18n.t("POST_COMMENT").POST_COMMENT_COMMENT}
                onPress={()=>{handleComment()}}
            />
            <View style={{alignItems:"flex-end"}}>
             <Icon style={{fontSize:18,color:"red"}} name='trash' onPress={()=>handleDelete(post.post_id)} />
             </View>
            <Card.Divider style={{marginTop:10}}/>
            <ScrollView>
            {comments?.map((comment,i)=>(
              <View key={i}>
                 <Text>- {comment.comment}</Text>
              </View>
            ))}
            </ScrollView>
           
            </Card>
        </StyledContainerStart>
           


    );
}

const styles = StyleSheet.create({
    image:{
      
        width: 300,
        height:200,
        marginBottom:10
      },

    buttonLike:{
      
      color: "#ffff",
      marginRight:10
      
     
    }  
  });