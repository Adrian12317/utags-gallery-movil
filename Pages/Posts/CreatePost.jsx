import React, { useEffect, useState } from 'react'
import { Text,StyleSheet,ActivityIndicator } from 'react-native';
import { Card , Button } from 'react-native-elements'
import CameraScreen from '../Camera/Camera';
import { storage } from '../../firebase';
import {
  StyledContainer,
  StyledInputCreate
} from './styledPostsComponent';
import Icon from 'react-native-vector-icons/FontAwesome';
import axios from 'axios';
import i18n from "../../localization/i18n"

export default function CreatePost({setIsCreate,getItems}) {
    const [isPhoto,setIsPhoto] = useState(false)
    const [isLoading,setIsLoading] = useState(false)
    const [image,setImage] = useState()
    const [nameFile,setNameFile] = useState()
    const [title,setTitle] = useState("")
    const [content,setContent] = useState("")
    const imageDefault = "https://www.smartdatajob.com/images/joomlart/demo/default.jpg";
    
    const handleSubmit = async () =>{
        
        let image_url = ""
        if (!title || !content) {
            
            return alert(i18n.t("POST_CREATE").POST_CREATE_ALERT_EMPTY);
        }
        setIsLoading((current) => current = true)
        
        if (image) {

            console.log("voy a guardar image");
            const response = await fetch(image);
            const blob = await response.blob();

            await storage.ref('posts/'+nameFile+'.jpg').put (blob).then (function (res) {
            
            }).catch(error => {
                console.log(error.message);
            }) 

            let image_url_1 = 'https://firebasestorage.googleapis.com/v0/b/login-adrian-44a42.appspot.com/o/posts%2F%20'+nameFile+'.jpg?alt=media'
            image_url = image_url_1.replace(/ /g, "")
        }else{
            image_url = imageDefault;
        }
        const request = {
            channel_id: 51,
            title: title,
            image_url: image_url,
            content: content
        }

       await axios.post('https://api-utagsgallery-codes.herokuapp.com/posts',request)
        .then(function ({data}) {
          // handle success
          setIsLoading((current) => current = false)
          console.log(data);
          setIsCreate(false);
          getItems()

        })
        .catch(function (error) {
          // handle error
          console.log(error);
            return alert("We have an error");
          
        })

       
    }


    return(
    <>
            {!isPhoto?(
            <StyledContainer>
                {isLoading?
                    <ActivityIndicator size="large" color="#00ff00" />
                    :
                    <Text></Text>
                }
                
                <Card>
                <Card.Title>{i18n.t("POST_CREATE").POST_CREATE_NAME}</Card.Title>
                <Card.Divider/>
                <Text>{i18n.t("POST_CREATE").POST_CREATE_TITLE}</Text>
                <StyledInputCreate onChangeText={setTitle} value={title}></StyledInputCreate>
                <Button
                 icon={<Icon style={styles.buttonLike} name='camera' />}
                buttonStyle={{width:300,marginBottom:5}}
                title={i18n.t("POST_CREATE").POST_CREATE_IMAGE}
                onPress={()=>{setIsPhoto(true)}}
                />
                {
                    image?
                    <Card.Image 
                    style={styles.image}
                    source={{uri:`${image}`}}
                    />            
                    :
                    <Text></Text>
                }
                
                
                <Text>{i18n.t("POST_CREATE").POST_CREATE_CONTENT}</Text>
                <StyledInputCreate
                    multiline={true}
                    numberOfLines={5}
                    onChangeText={setContent}
                    value={content}
                >
                </StyledInputCreate>
                <Card.Divider/>
                <Button
                buttonStyle={{width:300,backgroundColor:"#62ba24"}}
                title={i18n.t("POST_CREATE").POST_CREATE_CREATE}
                onPress={()=>{handleSubmit()}}
                />
                 <Button
                buttonStyle={{width:300,marginTop:10,backgroundColor:"#ff0000"}}
                title={i18n.t("POST_CREATE").POST_CREATE_CANCEL}
                onPress={()=>{setIsCreate(false)}}
                />
            </Card>
            </StyledContainer>
            ):
            (
                <CameraScreen
                 setTakePhoto={setIsPhoto}
                 setAvatar={setImage}
                 module="post"
                 setNameFile={setNameFile}
                 />
            )

            }
            </>


    );
}

const styles = StyleSheet.create({


    buttonLike:{
      
      color: "#ffff",
      marginRight:10
     
    }  
  });