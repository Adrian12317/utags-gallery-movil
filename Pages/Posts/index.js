export { default } from "./Posts.screen";