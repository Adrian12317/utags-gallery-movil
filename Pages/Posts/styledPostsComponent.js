import styled from 'styled-components/native';

export const StyledContainer = styled.View`
    flex: 1;
    align-items: center;
    background: #2A8D45;
    justify-content: center;
`;

export const StyledContainerStart = styled.View`
    flex: 1;
    align-items: flex-start;
    background: #2A8D45;
    justify-content: flex-start;
`;

export const StyledAddInput = styled.TextInput`     
    background: white;
    color: black;
    padding-left: 15;
    padding-right: 15;
    padding-top: 10;
    padding-bottom: 10;
    border-radius: 10;
    margin-top: 25;
    border-color: white;
    width: 70%;
`;

export const StyledInputCreate = styled.TextInput`     
    background: white;
    color: black;
    padding-bottom: 10;
    border-color: gray;
    width: 70%;
`;

export const StyledButton = styled.TouchableOpacity`
    background: #0782F9;
    width: 50%;
    padding-top: 15;
    padding-left: 15;
    padding-bottom: 15;
    padding-right: 15;
    border-radius: 10;
    align-items: center;
    margin-top: 15;
    margin-bottom: 10;
`;

export const StyledButtonText = styled.Text`
    color: white;
    font-weight: 700;
    font-size: 16;
`;

export const StyledScrollView = styled.ScrollView`
    background: #2A8D45;
    margin-bottom: 20;
    margin-left: 40;
    margin-top: 40;
    margin-right: 40;
`;



export const ItemsView = styled.View`
    flex-wrap: wrap;
    align-items: flex-start;
    flex-direction: row;
`;

export const StyledButtonDone = styled.Button`
    border-color: #1bf907;
    border-width: 3;
    border-radius: 100;
    margin-right: 5;
`;