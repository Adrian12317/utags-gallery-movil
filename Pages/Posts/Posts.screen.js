import React, { useCallback, useState } from 'react'
import { View,Text,StyleSheet,RefreshControl } from 'react-native';
import { useEffect } from 'react';
import { Card, SpeedDial , Button } from 'react-native-elements'
import {
  StyledContainer,
  StyledScrollView,
} from './styledPostsComponent';
import Icon from 'react-native-vector-icons/FontAwesome';
import CreatePost from './CreatePost';
import CommentPost from './CommentPost';
import i18n from "../../localization/i18n"

const axios = require('axios');

export default function Posts() {
  

    const [list,setList] = useState([])
    const [open,setOpen] = useState(false);
    const imageDefault = "https://www.smartdatajob.com/images/joomlart/demo/default.jpg";
    const [isCreate,setIsCreate] = useState(false);
    const [isComment,setIsComment] = useState(false);
    const [refreshing, setRefreshing] = useState(false);
    
    const onRefresh = useCallback(() => {
      setRefreshing(true);
      getItems()
    }, []);
    
    const [postId,setPostId] = useState() ;

    const getItems = () =>{
      axios.get('https://api-utagsgallery-codes.herokuapp.com/posts')
      .then(function ({data}) {
        // handle success
        setList([...data])
        setRefreshing(false)
      })
      .catch(function (error) {
        // handle error
        setList([])
        console.log(error);
        setRefreshing(false)
      })
    }

    const handleLike = async (_post_id) =>{
      axios.put('https://api-utagsgallery-codes.herokuapp.com/posts/like/'+_post_id)
      .then(function ({data}) {
        // handle success
        getItems();
      })
      .catch(function (error) {
        // handle error
        setList([])
        console.log(error);
      })
    }

    const handleComment = (_post_id) =>{
      setPostId(_post_id);
      setIsComment(true);
    }
    
    useEffect(()=>{
        getItems()
    },[])


    return(
     <StyledContainer>

       {isCreate?
        
          <CreatePost setIsCreate={setIsCreate} getItems={getItems} />
          :
          <></>
         }

         {isComment?
          <CommentPost setIsComment={setIsComment} post_id={postId} getItems={getItems}/>
         :
         <></>
         }

         {!isCreate & !isComment?
          <>
          <StyledScrollView
           refreshControl={
            <RefreshControl
              refreshing={refreshing}
              onRefresh={onRefresh}
            />
          }
          >
          
          {
          list?.map((x,i)=>(
    
            <Card  key={i}>
            <Card.Title>{x.title}</Card.Title>
            <Card.Divider/>
          
            <Card.Image  onPress={()=>{handleComment(x.post_id)}} style={styles.image} source={x.image_url? {uri:`${x.image_url}`}:{uri:imageDefault}}>             
            </Card.Image>
            <Text style={{marginBottom: 10}}>
                {x.content}
              </Text>
            <Button
                icon={<Icon style={styles.buttonLike} name='thumbs-up' />}
                buttonStyle={{width:70}}
                title={x.likes}
                onPress={()=>{handleLike(x.post_id)}}
              />
              <View style={styles.views}>
               <Text> <Icon style={styles.buttonView} name='eye' /> {x.views} </Text>
              </View>
            </Card>
          ))
              
          }
          </StyledScrollView>  
          <SpeedDial
            isOpen={open}
            icon={{ name: 'add', color: '#fff' }}
            openIcon={{ name: 'close', color: '#fff' }}
            onOpen={() => setOpen(!open)}
            onClose={() => setOpen(!open)}
            color="#316ef3"
            size="small"
          >
            <SpeedDial.Action
              icon={{ name: 'add', color: '#fff' }}
              title={i18n.t("POST").POST_NEW}
              onPress={() => {
                setIsCreate(true)
                setOpen(false)
              }}
              color="#316ef3"
              size="small"
            />
          </SpeedDial>
          </>
         :
         <></>
         }

  
  </StyledContainer>
        
  );
}

const styles = StyleSheet.create({
    buttonItemDone: {
      borderColor: "#1bf907",
      borderWidth: 3,
      borderRadius:100,
      marginRight:5
    },
    buttonItemRemove: {
        borderColor: "#f90707",
        borderWidth: 3,
        borderRadius:100
      },
    text: {
      color: "white",
      fontWeight: "700",
      fontSize: 18,
      marginTop:3
    },
    textDecoration: {
      color: "green",
      fontWeight: "700",
      fontSize: 18,
      marginTop:3,
      textDecorationLine:'line-through'
    },
    //nuevos para styled component
    image:{
      
      width: 250,
      height:200,
      marginBottom:10
    }  ,
    buttonLike:{
      
      color: "#ffff",
      marginRight:10
     
    } ,
    buttonView:{
      
      color: "#000",
      marginRight:10
     
    },
    views:{
      alignItems:"flex-end"
    }  
  });