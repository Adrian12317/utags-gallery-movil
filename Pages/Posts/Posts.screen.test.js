import React from "react";
import renderer from "react-test-renderer";
import Posts from "./Posts.screen";

describe("<Posts />", () => {
  it("has styled center", () => {
    const tree = renderer.create(<Posts />).toJSON();
    
    expect(tree.props.style[0].alignItems).toEqual("center");
  });
});