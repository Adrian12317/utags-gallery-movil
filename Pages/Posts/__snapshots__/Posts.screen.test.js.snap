// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#cf8538",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    style={
      Array [
        Object {
          "backgroundColor": "#cf8538",
          "marginBottom": 20,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 40,
        },
      ]
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "rgba(36, 36, 36, 0.6)",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "opacity": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginBottom": 16,
              "marginRight": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "color": "black",
                  "elevation": 2,
                  "marginHorizontal": 16,
                  "marginVertical": 8,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                },
                undefined,
              ]
            }
          >
            Add post
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 28,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  Object {
                    "borderRadius": 28,
                    "elevation": 4,
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#316ef3",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "padding": 8,
                      "width": 40,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 40,
                        "marginHorizontal": 5,
                        "overflow": "hidden",
                        "padding": 8,
                        "width": 40,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 16,
            "marginTop": 0,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              Object {
                "borderRadius": 28,
                "elevation": 4,
              },
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#316ef3",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 40,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 40,
                    "marginHorizontal": 5,
                    "overflow": "hidden",
                    "padding": 8,
                    "width": 40,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
